### PR TITLE
embed app now updates chunks with embeddings

### PR DESCRIPTION
**Pull request by gecBurton**
## Context

As a User I want the `embed` app to write the embedded text to ES so that it can be searched.

Following the discussion in the ticket and offline we have decided to update the existing `Chunk` which is already persisted to ES with the `embedding`.

## Changes proposed in this pull request

1. `Chunk` has a new optional `embedding field`
2. `EmbedQueueItem` now has a mandatory `chunk_uuid` field to enable us top identify the `Chunk` to update, but...
3. `EmbedQueueItem` has had the `sentence` removed so that it is only a reference to the `Chunk`
4. `ingest_file` now updates the relevant `Chunk` with the `embedding` field.

## Relevant links

https://technologyprogramme.atlassian.net/browse/REDBOX-93

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed

### CODE REVIEW

File: redbox/models/chat.py [hunk 1]
```diff
@@ -9,7 +9,6 @@
 
 
 class ChatMessage(PersistableModel):
-    uuid: str = Field(default_factory=lambda: str(uuid4()))
```
Comment (gecBurton) [inline redbox/models/chat.py:12]: not needed as field this is in the `PersistableModel` model

File: ingest/src/app.py [hunk 2]
```diff
@@ -45,7 +44,7 @@ def ingest_file(self, file: File):
         )
 
         file.processing_status = ProcessingStatusEnum.chunking
-        self.file_destination.update_item(file.uuid, file)
+        self.file_destination.update_item(str(file.uuid), file)
```
Comment (gecBurton) [inline ingest/src/app.py:47]:
```suggestion
        self.file_destination.update_item(file.uuid, file)
```

Comment (gecBurton) [inline ingest/src/app.py:47]:
```suggestion
        self.file_destination.update_item(file.uuid, file)
```

File: embed/src/app.py [hunk 3]
```diff
@@ -174,8 +180,17 @@ def embed_item_callback(ch: BlockingChannel, method, properties, body):
...
+        embedded_sentences = embed_sentences(embed_queue_item.model, [embed_queue_item.sentence])
+        logging.info(f"Embedding ID {embedded_sentences.embedding_id} complete for {method.delivery_tag}")
+
+        chunk: Chunk = storage_handler.read_item(str(embed_queue_item.chunk_uuid), "Chunk")
+
+        if len(embedded_sentences.data) != 1:
+            log.error(f"expected just 1 embedding but got {len(embedded_sentences.data)}")
+        else:
+            chunk.embedding = embedded_sentences.data[0].embedding
+            storage_handler.update_item(str(embed_queue_item.chunk_uuid), chunk)
```
Comment (gecBurton) [inline embed/src/app.py:192]:
```suggestion
            storage_handler.update_item(embed_queue_item.chunk_uuid, chunk)
```

File: embed/src/app.py [hunk 4]
```diff
@@ -174,8 +180,17 @@ def embed_item_callback(ch: BlockingChannel, method, properties, body):
...
         body_dict = json.loads(body.decode("utf-8"))
         embed_queue_item = EmbedQueueItem(**body_dict)
-        response = embed_sentences(embed_queue_item.model, [embed_queue_item.sentence])
-        logging.info(f"Embedding ID {response['embedding_id']} complete for {method.delivery_tag}")
+        embedded_sentences = embed_sentences(embed_queue_item.model, [embed_queue_item.sentence])
+        logging.info(f"Embedding ID {embedded_sentences.embedding_id} complete for {method.delivery_tag}")
+
+        chunk: Chunk = storage_handler.read_item(str(embed_queue_item.chunk_uuid), "Chunk")
+
+        if len(embedded_sentences.data) != 1:
```
Comment (gecBurton) [inline embed/src/app.py:188]: is this right?

File: embed/src/app.py [hunk 5]
```diff
@@ -174,8 +180,17 @@ def embed_item_callback(ch: BlockingChannel, method, properties, body):
     try:
         body_dict = json.loads(body.decode("utf-8"))
         embed_queue_item = EmbedQueueItem(**body_dict)
-        response = embed_sentences(embed_queue_item.model, [embed_queue_item.sentence])
-        logging.info(f"Embedding ID {response['embedding_id']} complete for {method.delivery_tag}")
+        embedded_sentences = embed_sentences(embed_queue_item.model, [embed_queue_item.sentence])
+        logging.info(f"Embedding ID {embedded_sentences.embedding_id} complete for {method.delivery_tag}")
+
+        chunk: Chunk = storage_handler.read_item(str(embed_queue_item.chunk_uuid), "Chunk")
```
Comment (gecBurton) [inline embed/src/app.py:186]:
```suggestion
        chunk: Chunk = storage_handler.read_item(embed_queue_item.chunk_uuid, "Chunk")
```

File: embed/src/app.py [hunk 6]
```diff
@@ -123,8 +129,8 @@ def get_model(model: str):
     return model_db.get_model_info(model)
 
 
-@app.post("/models/{model}/embed", response_model=EmbeddingResponse, tags=["models"])
-def embed_sentences(model: str, sentences: list[str]):
+@app.post("/models/{model}/embed", tags=["models"])
+def embed_sentences(model: str, sentences: list[str]) -> EmbeddingResponse:
```
Comment (gecBurton) [inline embed/src/app.py:133]: I have made the un-decorated function explicitly return `EmbeddingResponse` to help the case where this is called directly by `ingest_file`

File: redbox/models/llm.py [hunk 7]
```diff
@@ -38,5 +39,6 @@ class StatusResponse(BaseModel):
 
 
 class EmbedQueueItem(BaseModel):
-    model: str
-    sentence: str
```
Comment (gecBurton) [inline redbox/models/llm.py:42]: `sentence` replaced with `chunk_uuid`

**Review by lmwilkigov — APPROVED** [review]

**Pull request merged**